src/main/java/org/minions/devfund/chelo/Squarelotron.java

```diff
@@ -0,0 +1,183 @@
+        int counter = 1;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                this.squarelotron[i][j] = counter;
+                counter++;
+            }
+        }
+    }
+
+    /**
+     * A method to fill the 2-dimensional array with the numbers 1 to n squared.
+     * @param n size of the array
+     * @return array squarelotron filled
+     */
+    Squarelotron initializeArray(int n) {
+        Squarelotron array = new Squarelotron(n);
+        int counter = 1;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                array.squarelotron[i][j] = counter;
+                counter++;
+            }
+        }
+        return array;
+    }
+
+    /**
+     * A method to perform Upside-Down Flip of the squarelotron and returns the new squarelotron.
+     * The original squarelotron should not be modified.
+     * @param ring given to work on it
+     * @return new Squarelotron
+     */
+    Squarelotron upsideDownFlip(int ring) {
+        Squarelotron squarelotronAux = initializeArray(this.size);
+        squarelotronAux = getUpsideDownFlipArray(squarelotronAux, this.squarelotron, ring);
+        return squarelotronAux;
+    }
+
+    /**
+     * This method performs the Main Diagonal Flip of the squarelotron and returns the new squarelotron.
+     * @param ring the ring
+     * @return new array
+     */
+    Squarelotron mainDiagonalFlip(int ring) {
+        Squarelotron squarelotronDiagonal = initializeArray(this.size);
+
+        int indexX = ring - 1;
+        int aux;
+        for (int m = ring - 1; m < this.size - ring + 1; m++) {
+            aux = squarelotronDiagonal.squarelotron[indexX][ring - 1];
+            squarelotronDiagonal.squarelotron[indexX][ring - 1] = this.squarelotron[ring - 1][m];
+            squarelotronDiagonal.squarelotron[ring - 1][m] = aux;
+            indexX++;
+        }
+
+        indexX = ring;
+        for (int n = ring; n < this.size - ring; n++) {
+            aux = squarelotronDiagonal.squarelotron[indexX][this.size - ring];
+            squarelotronDiagonal.squarelotron[indexX][this.size - ring] = this.squarelotron[this.size - ring][n];
+            squarelotronDiagonal.squarelotron[this.size - ring][n] = aux;
+            indexX++;
+        }
+
+        return squarelotronDiagonal;
+    }
+
+    /**
+     * This method modifies the internal representation of the squarelotron.
+     * It does not create a new squarelotron.
+     * @param numberOfTurns The argument numberOfTurns indicates the number of times the entire
+     *                      squarelotron should be rotated 90° clockwise
+     */
+    void rotateRight(int numberOfTurns) {
+
+        final int majicTurns = 4;
+        Squarelotron arrayUpsideDown = new Squarelotron(this.size);
+        Squarelotron squarelotronRotate = new Squarelotron(this.size);
+        Squarelotron squarelotronAux = initializeArray(this.size);
```

Comment:
Don't create several instances of Squarelotron
edx unit tests are validating that.